[terra-framework-docs] Slider implementation guide and docs update (#1945)
cerner/terra-framework: packages/terra-framework-docs/src/terra-dev-site/doc/slider/About.1.doc.mdx

import { Badge } from 'terra-slider/package.json?dev-site-package';

import SliderPropsTable from 'terra-slider/src/Slider?dev-site-props-table';
import DefaultSlider from './Examples.2/DefaultSlider?dev-site-example';

<Badge />

# Terra Slider

## Table of Contents
* [Overview](/components/cerner-terra-framework-docs/slider/about#overview)
* [Examples](/components/cerner-terra-framework-docs/slider/about#examples)
* [Accessibility](/components/cerner-terra-framework-docs/slider/about#accessibility)
* [Implementation](/components/cerner-terra-framework-docs/slider/about#implementation)
* [Testing](/components/cerner-terra-framework-docs/slider/about#testing)
* [Properties](/components/cerner-terra-framework-docs/slider/about#properties)
* [Terra Standards](/components/cerner-terra-framework-docs/slider/about#terra-standards)

## Overview

Terra Slider provides a way to select a value from a given range by sliding a thumb along a bar to track the value.

> <Notice variant="caution" ariaLevel="2">

Reviewer: We need to stop using the Caution banner for these types of messages as I believe they require Reg Affairs approval. I think updating this to an Important! banner with a green background similar to what Charles did in the Terra Table documentation should work.

Author: Addressed all comments [here](https://github.com/cerner/terra-framework/pull/1945/commits/715f9208bff9a79fe767e7cc2d97decac2dff644 )

Terra Slider is an input component used to quickly select an imprecise value within a range such as a zoom level. It does not yet include features required for entering precise values.
</Notice>

<DefaultSlider />

### Features
Supports:

> * Management of custom slider label, minimum label and maximum label.

Reviewer: nit: Add an 'a' in front of custom
"Management of a custom slider lable, minimum label and maximum label."

> * Setting default value for slider.

Reviewer: Nit: Update it to "Setting a default value for the slider"

Reviewer: Can you add "the" in front of slider?

Author: Added

> * Disabling of slider.

Reviewer: Nit: Update it to "Disabling of the slider"

* Making other updates based on slider value change using the 'onChange' callback prop.
* Externally updating the slider value using the 'value' prop.

Refer to the [examples](/components/cerner-terra-framework-docs/slider/about#examples) section for detailed demonstration of the features supported by Slider.

> 

Reviewer 2: An Important block here explaining:

Consumers are given the ability to programmatically change the value of the slider without the user directly interacting with it. In most cases this isn't recommended. However, if this is done, ensure the UI makes it _very clear_ that the value was changed and that screen readers announce the change.

## Examples
|Link to Example| Description|
|-|-|
|[BasicSlider](/components/cerner-terra-framework-docs/slider/examples/default-slider)|An example of basic slider using the required props 'label', 'minimumValue', 'maximumValue', 'value'.|

> |[CustomSlider](/components/cerner-terra-framework-docs/slider/examples/custom-slider)|An example demonstrating a slider with a custom labels using the 'minimumLabel' and 'maximumLabel' props.|

Reviewer: Remove the "a" before "custom"

import { Badge } from 'terra-slider/package.json?dev-site-package';

import SliderPropsTable from 'terra-slider/src/Slider?dev-site-props-table';
import DefaultSlider from './Examples.2/DefaultSlider?dev-site-example';

<Badge />

# Terra Slider

## Table of Contents
* [Overview](/components/cerner-terra-framework-docs/slider/about#overview)
* [Examples](/components/cerner-terra-framework-docs/slider/about#examples)
* [Accessibility](/components/cerner-terra-framework-docs/slider/about#accessibility)
* [Implementation](/components/cerner-terra-framework-docs/slider/about#implementation)
* [Testing](/components/cerner-terra-framework-docs/slider/about#testing)
* [Properties](/components/cerner-terra-framework-docs/slider/about#properties)
* [Terra Standards](/components/cerner-terra-framework-docs/slider/about#terra-standards)

## Overview
<Notice variant="important" ariaLevel="2">

> The Terra Slider is an input component used to select an imprecise value within a range such as a zoom level. The component does not currently allow you to enter precise values.

Reviewer 2: ```suggestion
The Terra Slider is an input component used to select an imprecise value within a range such as a zoom level. The component does not currently support entry of precise values.
```
